### PR TITLE
Redirect to server root if login flow is started with no destination

### PR DESCRIPTION
**Pull request by liggitt**
Fixes https://github.com/openshift/origin-web-console/issues/888 and https://bugzilla.redhat.com/show_bug.cgi?id=1369854

### CODE REVIEW

**Comment by liggitt** [comment]
@juanvallejo @jwforres PTAL

File: pkg/auth/server/login/login.go [hunk 1]
```diff
@@ -152,6 +155,10 @@ func (l *Login) handleLogin(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 	then := req.FormValue("then")
+	if then == "" {
```
Comment (juanvallejo) [inline pkg/auth/server/login/login.go]: `len(then) == 0`?

**Comment by juanvallejo** [comment]
LGTM

**Comment by jwforres** [comment]
LGTM

**Comment by liggitt** [comment]
[merge]

**Comment by jwforres** [comment]
@liggitt looks like you need to update a test

**Pull request merged**